### PR TITLE
fix: adding one more pattern to relax the regex check for sts and impersonation url endpoints

### PR DESCRIPTION
**Pull request by BigTailWolf**
`sts-*.p.googleapis.com`

`iamcredentials-*.p.googleapis.com`

### CODE REVIEW

File: tests/test_external_account.py [hunk 1]
```diff
@@ -289,6 +289,7 @@ def test_valid_token_url_shall_pass_validation(self):
             "https://us-east-1-sts.googleapis.com",
             "https://US-WEST-1-sts.googleapis.com",
             "https://us-west-1-sts.googleapis.com/path?query",
+            "https://sts-us-east-1.p.googleapis.com",
```
Comment (aeitzman) [inline tests/test_external_account.py:292]: Add valid test case for url with /path added on end

Comment (aeitzman) [reply tests/test_external_account.py:292]: Also add test cases for invalid tests - 
            "https://sts.pgoogleapis.com",
            "https://p.googleapis.com",
            "https://sts.p.com",
            "http://sts.p.googleapis.com",
            "https://xyz-sts.p.googleapis.com",
            "https://sts-xyz.123.p.googleapis.com",
            "https://sts-xyz.p1.googleapis.com",
            "https://sts-xyz.p.foo.com",
            "https://sts-xyz.p.foo.googleapis.com");

Comment (lsirac) [inline tests/test_external_account.py:292]: Let's add some more test cases.

e.g:
https://github.com/googleapis/google-auth-library-nodejs/pull/1473

Comment (lsirac) [reply tests/test_external_account.py:292]: resolving; just saw Alex's comment

Comment (BigTailWolf) [reply tests/test_external_account.py:292]: Make sense

**Review by aeitzman — APPROVED** [review]

**Review by lsirac — APPROVED** [review]

**Review by clundin25 — APPROVED** [review]

**Review by sai-sunder-s — APPROVED** [review]

**Pull request merged**